lib/vagrant/action/builtin/cloud_init_wait.rb

```diff
@@ -0,0 +1,22 @@
+module Vagrant
+  module Action
+    module Builtin
+      class CloudInitWait
+
+        def initialize(app, env)
+          @app    = app
+          @logger = Log4r::Logger.new("vagrant::action::builtin::cloudinitwait")
+        end
+
+        def call(env)
+          machine = env[:machine]
+          @logger.info("Waiting for cloud init")
+          if !machine.config.vm.cloud_init_configs.empty?
+            machine.communicate.sudo("cloud-init status --wait", {:error_check => false})
```

Comment:
Hmm, I wonder if it's worth it to `Which::which` on the `cloud-init` executable just so we can be sure we're getting the right path on the guest.